### PR TITLE
[fix](function) fix date_format function execution error on fe

### LINKED ISSUE #31524: [Bug] function DATE_FORMAT get wrong result
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

2.0.5 && master

### What's Wrong?

case:
SELECT DATE_FORMAT('1999-01-01', '%X %V')

wrong result:
```
MySQL [test]> SELECT DATE_FORMAT('1999-01-01', '%X %V');
+-----------------------------------------------------------+
| date_format(cast('1999-01-01' as DATETIMEV2(0)), '%X %V') |
+-----------------------------------------------------------+
| 1999 01                                                   |
+-----------------------------------------------------------+
1 row in set (0.012 sec)
```

mysql right result:
```
mysql> SELECT DATE_FORMAT('1999-01-01', '%X %V');
+------------------------------------+
| DATE_FORMAT('1999-01-01', '%X %V') |
+------------------------------------+
| 1998 52                            |
+------------------------------------+
1 row in set (0.00 sec)
```

### What You Expected?

result same as mysql

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by felixwluo**
## Proposed changes

Issue Number: close #31524

`%X and %V are formatted according to Sunday as the first day of the week. After repair, sql: select date_format('1999-01-01 00:00:00', '%X %V'); The result is 1998 52`

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by felixwluo** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/util/DateUtils.java [hunk 1]
```diff
@@ -39,6 +42,7 @@ public class DateUtils {
      */
     public static DateTimeFormatterBuilder formatBuilder(String pattern) throws AnalysisException {
         DateTimeFormatterBuilder builder = new DateTimeFormatterBuilder();
+        WeekFields weekFields;
```
Comment (jackwener) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/util/DateUtils.java:45]: Remove it

File: fe/fe-core/src/main/java/org/apache/doris/nereids/util/DateUtils.java [hunk 2]
```diff
@@ -103,7 +107,8 @@ public static DateTimeFormatterBuilder formatBuilder(String pattern) throws Anal
                         builder.appendPattern("HH:mm:ss");
                         break;
                     case 'V': // %V Week (01..53), where Sunday is the first day of the week; used with %X
-                        builder.appendValue(ChronoField.ALIGNED_WEEK_OF_YEAR, 2);
+                        weekFields = WeekFields.of(DayOfWeek.SUNDAY, 7);
+                        builder.appendValue(weekFields.weekOfWeekBasedYear(), 2);
```
[the remark this answers is not in the document]
Comment (felixwluo) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/util/DateUtils.java:111]: I need to use weekOfWeekBasedYear to get the number of weeks in a week-based year

File: fe/fe-core/src/main/java/org/apache/doris/nereids/util/DateUtils.java (hunk 1 above)
Comment (jackwener) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/util/DateUtils.java:45]: Add 

```
    private static WeekFields  weekFields = WeekFields.of(DayOfWeek.SUNDAY, 7);;
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/util/DateUtils.java [hunk 3]
```diff
@@ -103,7 +107,8 @@ public static DateTimeFormatterBuilder formatBuilder(String pattern) throws Anal
                         builder.appendPattern("HH:mm:ss");
                         break;
                     case 'V': // %V Week (01..53), where Sunday is the first day of the week; used with %X
-                        builder.appendValue(ChronoField.ALIGNED_WEEK_OF_YEAR, 2);
+                        weekFields = WeekFields.of(DayOfWeek.SUNDAY, 7);
```
Comment (jackwener) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/util/DateUtils.java:110]: remove it

File: fe/fe-core/src/main/java/org/apache/doris/nereids/util/DateUtils.java [hunk 4]
```diff
@@ -115,6 +120,9 @@ public static DateTimeFormatterBuilder formatBuilder(String pattern) throws Anal
                         builder.appendValue(IsoFields.WEEK_BASED_YEAR, 4);
                         break;
                     case 'X':
+                        weekFields = WeekFields.of(DayOfWeek.SUNDAY, 7);
```
Comment (jackwener) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/util/DateUtils.java:123]: remove it

Comment (felixwluo) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/util/DateUtils.java]: Thank you. It's been fixed

**Comment by cambyzju** [comment]
run buildall

**Review by jackwener — DISMISSED** [review]

**Review by zclllyybb — APPROVED** [review]
LGTM for code. Remember to change the fe-ut result.

**Review by zclllyybb — APPROVED** [review]

**Review by cambyzju — APPROVED** [review]

**Comment by zclllyybb** [comment]
run buildall

**Comment by xiaokang** [comment]
@felixwluo Can you cherry-pick this pr to branch-2.0? You can use the script tools/pick_pr.sh to do it in 1 minute.

**Comment by felixwluo** [comment]
> @felixwluo Can you cherry-pick this pr to branch-2.0? You can use the script tools/pick_pr.sh to do it in 1 minute.

okay

**Comment by felixwluo** [comment]
> @felixwluo Can you cherry-pick this pr to branch-2.0? You can use the script tools/pick_pr.sh to do it in 1 minute.

https://github.com/apache/doris/pull/31921

**Pull request merged**